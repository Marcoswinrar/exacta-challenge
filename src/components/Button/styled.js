import styled from 'styled-components'
import media from 'styled-media-query'

export const ButtonWrapper = styled.button`
  width: ${props => props.width ? props.width : '145px'};
  font-size: 1.2rem;
  border: none;
  font-weight: bold;
  text-align: center;
  text-transform: uppercase;
  padding: 10px;
  cursor: pointer;
  box-shadow: 0 1px #999;
  background-color: #22a6b3;
  color: white;
  
  :active {
    box-shadow: 0 3px #666;
    transform: translateY(4px);
  }

  ${media.lessThan("320px")`
    width: 120px;
    font-size: 1rem;
  `}
`