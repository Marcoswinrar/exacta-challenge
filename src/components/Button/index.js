import * as S from './styled'

const Button = ({ width, value, onClick, active, ...props }) => {
  return (
    <S.ButtonWrapper
      {...props}
      style={{ backgroundColor: active && '#22a6b3'}}
      width={width}
      onClick={onClick}>
      {value}
    </S.ButtonWrapper>
  )
}

export default Button