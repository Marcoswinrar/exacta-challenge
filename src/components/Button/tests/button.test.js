import { fireEvent, render, screen } from "@testing-library/react"
import { act } from "react-dom/test-utils"
import Button from ".."

describe("Componente <Button />", () => {
  
  it("Deve renderizar o button sem problemas", () => {
    const {container} = render(<Button value="Botão" />)

    expect(container).toMatchSnapshot()
    expect(screen.queryByText(/Botão/)).toBeInTheDocument()
  })

  it("Deve renderizar o botão com o tamanho que for passado", () => {
    render(<Button width={"145px"}/>)

    const botao = screen.getByRole("button")

    expect(botao).toHaveStyle({
      "width": "145px"
    })
  })

  it("Deve chamar função de click ao clicar no botão", () => {
    const onClick = jest.fn()

    render(<Button value="Botão" onClick={onClick}/>)

    const botao = screen.queryByText(/Botão/)

    act(() => fireEvent.click(botao))

    expect(onClick).toBeCalledTimes(1)

  })
})