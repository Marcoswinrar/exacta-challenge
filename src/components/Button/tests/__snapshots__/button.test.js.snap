// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Button /> Deve renderizar o button sem problemas 1`] = `
<div>
  <button
    class="sc-bczRLJ gtaYVB"
  >
    Bot√£o
  </button>
</div>
`;
